src/components/modal/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, PropsWithChildren } from 'react';
+import {
+  Modal as ChakraModal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  ModalOverlayProps,
+  ModalProps,
+} from '@chakra-ui/react';
+
+import Divider from '../divider';
+import Button from '../button';
+
+type ActionButton = {
+  action: () => void;
+  label: string;
+};
+
+interface Props extends ModalProps {
+  title?: string;
+  modalOverlayProps?: ModalOverlayProps;
```

Comment:
as discussed, I would not expose those. But I'll leave that up to you to decide. Optional